from PyQt5 import QtCore, QtGui, QtWidgets
import pymysql
import main5


class Ui_resever(object):
    def setupUi(self, resever):
        resever.setObjectName("resever")
        resever.setWindowModality(QtCore.Qt.NonModal)
        resever.resize(1280, 720)
        resever.setStyleSheet("\n"
"background-color: rgba(47, 47, 47, 220);")
        resever.setToolButtonStyle(QtCore.Qt.ToolButtonTextUnderIcon)
        resever.setDocumentMode(False)
        self.centralwidget = QtWidgets.QWidget(resever)
        self.centralwidget.setObjectName("centralwidget")
        self.RESEVER = QtWidgets.QLabel(self.centralwidget)
        self.RESEVER.setGeometry(QtCore.QRect(150, -10, 241, 151))
        font = QtGui.QFont()
        font.setFamily("Mali Medium")
        font.setPointSize(36)
        self.RESEVER.setFont(font)
        self.RESEVER.setStyleSheet("color: rgb(255, 255, 255);")
        self.RESEVER.setObjectName("RESEVER")
        self.photo = QtWidgets.QLabel(self.centralwidget)
        self.photo.setGeometry(QtCore.QRect(530, 20, 691, 451))
        self.photo.setText("")
        self.photo.setPixmap(QtGui.QPixmap("C:/Users/ASUS/OneDrive/เดสก์ท็อป/สไลด์1.PNG"))
        self.photo.setScaledContents(True)
        self.photo.setObjectName("photo")
        self.nameTEXT = QtWidgets.QLabel(self.centralwidget)
        self.nameTEXT.setGeometry(QtCore.QRect(120, 160, 81, 51))
        font = QtGui.QFont()
        font.setFamily("Mali Medium")
        font.setPointSize(16)
        self.nameTEXT.setFont(font)
        self.nameTEXT.setStyleSheet("color: rgb(255, 255, 255);")
        self.nameTEXT.setObjectName("nameTEXT")
        self.name = QtWidgets.QLineEdit(self.centralwidget)
        self.name.setGeometry(QtCore.QRect(220, 170, 211, 31))
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Expanding, QtWidgets.QSizePolicy.Fixed)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.name.sizePolicy().hasHeightForWidth())
        self.name.setSizePolicy(sizePolicy)
        self.name.setContextMenuPolicy(QtCore.Qt.DefaultContextMenu)
        self.name.setStyleSheet("background-color: rgb(255, 255, 255);")
        self.name.setText("")
        self.name.setCursorMoveStyle(QtCore.Qt.LogicalMoveStyle)
        self.name.setObjectName("name")
        self.teltext = QtWidgets.QLabel(self.centralwidget)
        self.teltext.setGeometry(QtCore.QRect(120, 230, 81, 51))
        font = QtGui.QFont()
        font.setFamily("Mali Medium")
        font.setPointSize(16)
        self.teltext.setFont(font)
        self.teltext.setStyleSheet("color: rgb(255, 255, 255);")
        self.teltext.setObjectName("teltext")
        self.tel = QtWidgets.QLineEdit(self.centralwidget)
        self.tel.setGeometry(QtCore.QRect(220, 240, 211, 31))
        self.tel.setStyleSheet("background-color: rgb(255, 255, 255);")
        self.tel.setObjectName("tel")
        self.locationtext = QtWidgets.QLabel(self.centralwidget)
        self.locationtext.setGeometry(QtCore.QRect(120, 300, 221, 51))
        font = QtGui.QFont()
        font.setFamily("Mali Medium")
        font.setPointSize(16)
        self.locationtext.setFont(font)
        self.locationtext.setStyleSheet("color: rgb(255, 255, 255);")
        self.locationtext.setObjectName("locationtext")
        self.Clientlisttext = QtWidgets.QLabel(self.centralwidget)
        self.Clientlisttext.setGeometry(QtCore.QRect(530, 480, 141, 51))
        font = QtGui.QFont()
        font.setFamily("Mali Medium")
        font.setPointSize(16)
        self.Clientlisttext.setFont(font)
        self.Clientlisttext.setStyleSheet("color: rgb(255, 255, 255);")
        self.Clientlisttext.setObjectName("Clientlisttext")
        self.editlist = QtWidgets.QPushButton(self.centralwidget)
        self.editlist.setGeometry(QtCore.QRect(1050, 490, 75, 21))
        self.editlist.setStyleSheet("background-color: rgb(241, 161, 0);\n"
"font: 57 10pt \"Mali Medium\";\n"
"color: rgb(255, 255, 255);\n"
"border-radius:5;")
        self.editlist.setObjectName("editlist")
        self.total = QtWidgets.QLabel(self.centralwidget)
        self.total.setGeometry(QtCore.QRect(120, 440, 311, 81))
        font = QtGui.QFont()
        font.setFamily("Mali Medium")
        font.setPointSize(28)
        self.total.setFont(font)
        self.total.setStyleSheet("color: rgb(255, 255, 255);")
        self.total.setObjectName("total")
        self.save = QtWidgets.QPushButton(self.centralwidget)
        self.save.setGeometry(QtCore.QRect(80, 570, 161, 41))
        font = QtGui.QFont()
        font.setFamily("Mali Medium")
        font.setPointSize(16)
        self.save.setFont(font)
        self.save.setStyleSheet("background-color: rgb(0, 243, 0);\n"
"color: rgb(255, 255, 255);\n"
"border-radius:5;")
        self.save.setObjectName("save")
        self.reset = QtWidgets.QPushButton(self.centralwidget)
        self.reset.setGeometry(QtCore.QRect(310, 570, 161, 41))
        font = QtGui.QFont()
        font.setFamily("Mali Medium")
        font.setPointSize(16)
        self.reset.setFont(font)
        self.reset.setStyleSheet("background-color: rgb(255, 0, 0);\n"
"color: rgb(255, 255, 255);\n"
"border-radius:5;")
        self.reset.setObjectName("reset")
        self.zone_2 = QtWidgets.QComboBox(self.centralwidget)
        self.zone_2.setGeometry(QtCore.QRect(350, 310, 81, 31))
        self.zone_2.setStyleSheet("background-color: rgb(255, 255, 255);")
        self.zone_2.setFrame(True)
        self.zone_2.setObjectName("zone_2")
        self.zone_2.addItem("")
        self.zone_2.setItemText(0, "")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.zone_2.addItem("")
        self.tableWidget = QtWidgets.QTableWidget(self.centralwidget)
        self.tableWidget.setGeometry(QtCore.QRect(530, 520, 691, 141))
        self.tableWidget.setStyleSheet("background-color: rgba(255, 255, 255, 220);")
        self.tableWidget.setShowGrid(True)
        self.tableWidget.setRowCount(0)
        self.tableWidget.setObjectName("tableWidget")
        self.tableWidget.setColumnCount(4)
        item = QtWidgets.QTableWidgetItem()
        self.tableWidget.setHorizontalHeaderItem(0, item)
        item = QtWidgets.QTableWidgetItem()
        self.tableWidget.setHorizontalHeaderItem(1, item)
        item = QtWidgets.QTableWidgetItem()
        self.tableWidget.setHorizontalHeaderItem(2, item)
        item = QtWidgets.QTableWidgetItem()
        self.tableWidget.setHorizontalHeaderItem(3, item)
        self.tableWidget.horizontalHeader().setMinimumSectionSize(39)
        self.tableWidget.verticalHeader().setDefaultSectionSize(40)
        self.tableWidget.verticalHeader().setMinimumSectionSize(30)
        self.pushButton = QtWidgets.QPushButton(self.centralwidget)
        self.pushButton.setGeometry(QtCore.QRect(220, 380, 111, 41))
        self.pushButton.setStyleSheet("background-color: rgb(71, 214, 0);\n"
"color: rgb(255, 255, 255);\n"
"font: 8pt \"MS Shell Dlg 2\";\n"
"font: 57 16pt \"Mali Medium\";\n"
"border-radius:5;")
        self.pushButton.setObjectName("pushButton")
        self.editlist_2 = QtWidgets.QPushButton(self.centralwidget)
        self.editlist_2.setGeometry(QtCore.QRect(1150, 490, 75, 21))
        self.editlist_2.setStyleSheet("background-color: rgb(255, 170, 127);\n"
"font: 57 10pt \"Mali Medium\";\n"
"color: rgb(255, 255, 255);\n"
"border-radius:5;")
        self.editlist_2.setObjectName("editlist_2")
        self.Clientlisttext.raise_()
        self.RESEVER.raise_()
        self.photo.raise_()
        self.nameTEXT.raise_()
        self.name.raise_()
        self.teltext.raise_()
        self.tel.raise_()
        self.locationtext.raise_()
        self.editlist.raise_()
        self.total.raise_()
        self.save.raise_()
        self.reset.raise_()
        self.zone_2.raise_()
        self.tableWidget.raise_()
        self.pushButton.raise_()
        self.editlist_2.raise_()
        resever.setCentralWidget(self.centralwidget)
        self.menubar = QtWidgets.QMenuBar(resever)
        self.menubar.setGeometry(QtCore.QRect(0, 0, 1280, 21))
        self.menubar.setObjectName("menubar")
        resever.setMenuBar(self.menubar)
        self.statusbar = QtWidgets.QStatusBar(resever)
        self.statusbar.setObjectName("statusbar")
        resever.setStatusBar(self.statusbar)

        self.retranslateUi(resever)
        QtCore.QMetaObject.connectSlotsByName(resever)

    def retranslateUi(self, resever):
        _translate = QtCore.QCoreApplication.translate
        resever.setWindowTitle(_translate("resever", "MainWindow"))
        self.RESEVER.setText(_translate("resever", "<html><head/><body><p align=\"center\">RESEVER</p></body></html>"))
        self.nameTEXT.setText(_translate("resever", "<html><head/><body><p><span style=\" font-weight:600;\">NAME : </span></p></body></html>"))
        self.teltext.setText(_translate("resever", "<html><head/><body><p><span style=\" font-weight:600;\">TEL : </span></p></body></html>"))
        self.locationtext.setText(_translate("resever", "<html><head/><body><p><span style=\" font-weight:600;\">ZONE/LOCATION : </span></p></body></html>"))
        self.Clientlisttext.setText(_translate("resever", "<html><head/><body><p><span style=\" font-weight:600;\">Client lists </span></p></body></html>"))
        self.editlist.setText(_translate("resever", "Edit List"))
        self.total.setText(_translate("resever", "<html><head/><body><p><span style=\" font-size:26pt;\">TOTAL 0.00 BAHT</span></p></body></html>"))
        self.save.setText(_translate("resever", "SAVE"))
        self.reset.setText(_translate("resever", "RESET"))
        self.zone_2.setItemText(1, _translate("resever", "A1 / 500"))
        self.zone_2.setItemText(2, _translate("resever", "A2 / 500"))
        self.zone_2.setItemText(3, _translate("resever", "A3 / 500"))
        self.zone_2.setItemText(4, _translate("resever", "A4 / 500"))
        self.zone_2.setItemText(5, _translate("resever", "A5 / 500"))
        self.zone_2.setItemText(6, _translate("resever", "A6 / 500"))
        self.zone_2.setItemText(7, _translate("resever", "A7 / 500"))
        self.zone_2.setItemText(8, _translate("resever", "A8 / 500"))
        self.zone_2.setItemText(9, _translate("resever", "A9 / 500"))
        self.zone_2.setItemText(10, _translate("resever", "A10 / 500"))
        self.zone_2.setItemText(11, _translate("resever", "A11 / 500"))
        self.zone_2.setItemText(12, _translate("resever", "A12 / 500"))
        self.zone_2.setItemText(13, _translate("resever", "A13 / 500"))
        self.zone_2.setItemText(14, _translate("resever", "A14 / 500"))
        self.zone_2.setItemText(15, _translate("resever", "A15 / 500"))
        self.zone_2.setItemText(16, _translate("resever", "A16 / 500"))
        self.zone_2.setItemText(17, _translate("resever", "B1 / 400"))
        self.zone_2.setItemText(18, _translate("resever", "B2 / 400"))
        self.zone_2.setItemText(19, _translate("resever", "B3 / 400"))
        self.zone_2.setItemText(20, _translate("resever", "B4 / 400"))
        self.zone_2.setItemText(21, _translate("resever", "B5 / 400"))
        self.zone_2.setItemText(22, _translate("resever", "B6 / 400"))
        self.zone_2.setItemText(23, _translate("resever", "B7 / 400"))
        self.zone_2.setItemText(24, _translate("resever", "B8 / 400"))
        self.zone_2.setItemText(25, _translate("resever", "B9 / 400"))
        self.zone_2.setItemText(26, _translate("resever", "B10 / 400"))
        self.zone_2.setItemText(27, _translate("resever", "B11 / 400"))
        self.zone_2.setItemText(28, _translate("resever", "B12 / 400"))
        self.zone_2.setItemText(29, _translate("resever", "B13 / 400"))
        self.zone_2.setItemText(30, _translate("resever", "B14 / 400"))
        self.zone_2.setItemText(31, _translate("resever", "B15 / 400"))
        self.zone_2.setItemText(32, _translate("resever", "C1 / 400"))
        self.zone_2.setItemText(33, _translate("resever", "C2 / 400"))
        self.zone_2.setItemText(34, _translate("resever", "C3 / 400"))
        self.zone_2.setItemText(35, _translate("resever", "C4 / 400"))
        self.zone_2.setItemText(36, _translate("resever", "C5 / 400"))
        self.zone_2.setItemText(37, _translate("resever", "C6 / 400"))
        self.zone_2.setItemText(38, _translate("resever", "C7 / 400"))
        self.zone_2.setItemText(39, _translate("resever", "C8 / 400"))
        self.zone_2.setItemText(40, _translate("resever", "C9 / 400"))
        self.zone_2.setItemText(41, _translate("resever", "C10 / 400"))
        self.zone_2.setItemText(42, _translate("resever", "C11 / 400"))
        self.zone_2.setItemText(43, _translate("resever", "C12 / 400"))
        self.zone_2.setItemText(44, _translate("resever", "C13 / 400"))
        self.zone_2.setItemText(45, _translate("resever", "C14 / 400"))
        self.zone_2.setItemText(46, _translate("resever", "C15 / 400"))
        self.zone_2.setItemText(47, _translate("resever", "D1 / 300"))
        self.zone_2.setItemText(48, _translate("resever", "D2 / 300"))
        self.zone_2.setItemText(49, _translate("resever", "D3 / 300"))
        self.zone_2.setItemText(50, _translate("resever", "D4 / 300"))
        self.zone_2.setItemText(51, _translate("resever", "D5 / 300"))
        self.zone_2.setItemText(52, _translate("resever", "D6 / 300"))
        self.zone_2.setItemText(53, _translate("resever", "D7 / 300"))
        self.zone_2.setItemText(54, _translate("resever", "D8 / 300"))
        self.zone_2.setItemText(55, _translate("resever", "D9 / 300"))
        self.zone_2.setItemText(56, _translate("resever", "D10 / 300"))
        self.zone_2.setItemText(57, _translate("resever", "D11 / 300"))
        self.zone_2.setItemText(58, _translate("resever", "D12 / 300"))
        self.zone_2.setItemText(59, _translate("resever", "D13 / 300"))
        self.zone_2.setItemText(60, _translate("resever", "D14 / 300"))
        self.zone_2.setItemText(61, _translate("resever", "D15 / 300"))
        self.zone_2.setItemText(62, _translate("resever", "D16 / 300"))
        self.zone_2.setItemText(63, _translate("resever", "D17 / 300"))
        self.zone_2.setItemText(64, _translate("resever", "D18 / 300"))
        self.zone_2.setItemText(65, _translate("resever", "D19 / 300"))
        self.zone_2.setItemText(66, _translate("resever", "D20 / 300"))
        self.zone_2.setItemText(67, _translate("resever", "D21 / 300"))
        self.zone_2.setItemText(68, _translate("resever", "D22 / 300"))
        self.zone_2.setItemText(69, _translate("resever", "D23 / 300"))
        self.zone_2.setItemText(70, _translate("resever", "D24 / 300"))
        item = self.tableWidget.horizontalHeaderItem(0)
        item.setText(_translate("resever", "ID"))
        item = self.tableWidget.horizontalHeaderItem(1)
        item.setText(_translate("resever", "name"))
        item = self.tableWidget.horizontalHeaderItem(2)
        item.setText(_translate("resever", "tel"))
        item = self.tableWidget.horizontalHeaderItem(3)
        item.setText(_translate("resever", "zone"))
        self.pushButton.setText(_translate("resever", "SUM"))
        self.editlist_2.setText(_translate("resever", "Refresh"))
        self.save.clicked.connect(self.insertDBS)
        self.save.clicked.connect(self.refresh)
        self.reset.clicked.connect(self.cleartext)
        self.pushButton.clicked.connect(self.calculate)
        self.editlist_2.clicked.connect(self.refresh)
        self.editlist.clicked.connect(self.openedit)

    
    def calculate(self):
         price= int(self.zone_2.currentText().split('/')[1].strip())
         self.total.setText('TOTAL '+ str(price) +' BAHT')
   

    def insertDBS(self):
        names = self.name.text()
        tels = self.tel.text()
        zones = str(self.zone_2.currentText().split('/')[0])
        if ( names == '' or tels == '' or zones == '' ):
            print("Please fill all data")
        else:
            con = pymysql.connect(host="localhost", user='root', password='', database="gewksix")
            cursor = con.cursor()
            cursor.execute("INSERT INTO resever (name, tel, zone) values (%s, %s, %s)",(names, tels, zones ))
            con.commit()
            self.name.setText("")
            self.tel.setText("")
            self.zone_2.setCurrentText("")
            
    def cleartext(self):  
        self.name.setText("")
        self.tel.setText("")
        self.zone_2.setCurrentText("")

    def refresh(self):
        print('refreshed data in the table')
        self.tableWidget.setRowCount(0)
        con = pymysql.connect(host="localhost", database="gewksix",user='root', password='')
        cursor = con.cursor()
        cursor.execute("SELECT ID,name,tel,zone FROM resever")
        data = list(cursor.fetchall())
        tablerow = 0
        for i in data:
            self.tableWidget.insertRow(tablerow)
            self.tableWidget.setItem(tablerow, 0, QtWidgets.QTableWidgetItem(str(i[0])))
            self.tableWidget.setItem(tablerow, 1, QtWidgets.QTableWidgetItem(i[1]))
            self.tableWidget.setItem(tablerow, 2, QtWidgets.QTableWidgetItem(str(i[2])))
            self.tableWidget.setItem(tablerow, 3, QtWidgets.QTableWidgetItem(i[3]))
            tablerow += 1

    def openedit(self):
        self.window = QtWidgets.QMainWindow()
        self.ui = main5.Ui_Main()
        self.ui.setupUi(self.window)
        self.window.show()


if __name__ == "__main__":
    import sys
    app = QtWidgets.QApplication(sys.argv)
    resever = QtWidgets.QMainWindow()
    ui = Ui_resever()
    ui.setupUi(resever)
    resever.show()
    sys.exit(app.exec_())
